caravel/views.py

```diff
@@ -1042,45 +1042,64 @@ def request_access(self):
     @has_access
     @expose("/approve")
     def approve(self):
-        request_access_id = request.args.get('request_access_id')
+        datasource_type = request.args.get('datasource_type')
+        datasource_id = request.args.get('datasource_id')
+        created_by_username = request.args.get('created_by')
         role_to_grant = request.args.get('role_to_grant')
         role_to_extend = request.args.get('role_to_extend')
-        access_request = (
-            db.session.query(models.DatasourceAccessRequest)
-            .filter_by(id=request_access_id).first()
-        )
+
+        session = db.session
+        datasource_class = src_registry.sources[datasource_type]
+        datasource = session.query(datasource_class).filter_by(
+            id=datasource_id).first()
+
+        if not datasource:
+            flash(DATASOURCE_MISSING_ERR, "alert")
+            return json_error_response(DATASOURCE_MISSING_ERR)
+
+        requested_by = sm.find_user(username=created_by_username)
+        if not requested_by:
+            flash(USER_MISSING_ERR, "alert")
+            return json_error_response(USER_MISSING_ERR)
+
+        requests = (session.query(models.DatasourceAccessRequest).filter(
```

Comment:
optional: indent is inconsistent with other queries, looks like maybe sublime auto-indented that